homeassistant/components/tado/water_heater.py

```diff
@@ -212,6 +234,15 @@ def set_operation_mode(self, operation_mode):
 
         self._control_heater(hvac_mode=mode)
 
+    def set_timer(self, time_period, temperature=None):
+        """Set the timer on the entity, and temperature if supported."""
+        if not self._supports_temperature_control and temperature is not None:
```

Comment:
Just wondering, why return and stop? Why not continue with just the `timer_period`?
This prevents setting a timer on a mixed group of devices (with ones that do and don't support temperature).

With this case, some will silently fail on setting timer, which feel a bit odd?

Reply:
I changed it off the back of an earlier review comment!  I can change back, but does @bdraco need to agree?

Reply:
It should probably throw an exception if you ask for a temperature and the device doesn't support it.

Reply:
that would mean in frencks example if you were setting the timer on a mixed group you would get an exception on some but not all of them.  Is that ok?

Reply:
If any of the input is invalid we should throw an exception.  It is too messy to try to handle partial failures since we don't have a good way to report individual result states.  The user would likely be confused as to which device did what they expected and which device didn't if we silently dropped invalid inputs.

Reply:
I think it is fine to simply not pass the temperature in case the device doesn't support that.

This is in line with how we handle it in order places. E.g., when turning on lights in an area, with a color and brightness, it would pass successfully if a light doesn't support color or brightness without an error.

Reply:
I'm a subscriber to fail as quickly as possible if there is any bad input, and I spend a good chunk of my week rejecting code with security issues from accepting unexpected user input.  Home Assistant doesn't really have those same security concerns since the user isn't generally considered potentially hostile so this isn't something to get stuck on.  

@jfearon Let's get you unblocked on this and go with @frenck 's suggestion.

Reply:
Set temperature to none now when it is not supported

Reply:
It looks like the change didn't get pushed?

Reply:
Not sure if i messed something up, but for me line 237 has:
 temperature = None 
whereas it was a return before?

Reply:
I see it now